cmd/mattermost/commands/db.go

```diff
@@ -133,20 +154,103 @@ func resetCmdF(command *cobra.Command, args []string) error {
 
 func migrateCmdF(command *cobra.Command, args []string) error {
 	cfgDSN := getConfigDSN(command, config.GetEnvironment())
+	recoverFlag, _ := command.Flags().GetBool("auto-recover")
+	savePlan, _ := command.Flags().GetBool("save-plan")
+	dryRun, _ := command.Flags().GetBool("dry-run")
 	cfgStore, err := config.NewStoreFromDSN(cfgDSN, true, nil, true)
 	if err != nil {
 		return errors.Wrap(err, "failed to load configuration")
 	}
 	config := cfgStore.Get()
 
-	store := sqlstore.New(config.SqlSettings, nil)
-	defer store.Close()
+	plan, err := sqlstore.GeneratePlan(config.SqlSettings, recoverFlag)
```

Comment:
One issue that I see is that we are doing `ss.initConnection()` twice: once with `GeneratePlan`, and again with `MigrateWithPlan`. We should just create the connection once and then run methods.